### PR TITLE
[Ink] Combine initialization method for MDCLegacyInkLayer

### PR DESCRIPTION
**Pull request by maicki**

### CODE REVIEW

File: components/Ink/src/private/MDCLegacyInkLayer.m [hunk 1]
```diff
@@ -490,24 +484,29 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
...
-    _foregroundRipples = [NSMutableArray array];
-    _backgroundRipples = [NSMutableArray array];
-    [self addSublayer:_compositeRipple];
+    [self _initLegacyInkLayer];
   }
 
   return self;
 }
 
+- (void)_initLegacyInkLayer {
```
Comment (jverkoey) [inline components/Ink/src/private/MDCLegacyInkLayer.m]: Our code convention for common initializers is `common<#ClassName#>Init`. This helps reduce the risk of subclasses colliding with superclass common inits.

In this case then: `commonMDCLegacyInkLayerInit`

**Review by jverkoey — CHANGES_REQUESTED** [review]

File: components/Ink/src/private/MDCLegacyInkLayer.m (hunk 1 above)
Comment (maicki) [reply components/Ink/src/private/MDCLegacyInkLayer.m]: Thanks for pointing out. Changed.

**Review by jverkoey — APPROVED** [review]

File: components/Ink/src/private/MDCLegacyInkLayer.m [hunk 2]
```diff
@@ -475,13 +475,7 @@ @implementation MDCLegacyInkLayer
 - (instancetype)init {
   self = [super init];
   if (self) {
-    self.masksToBounds = YES;
```
Comment (yarneo) [inline components/Ink/src/private/MDCLegacyInkLayer.m]: it looks like we have lost this masksToBounds = YES with this change.

**Review by yarneo — CHANGES_REQUESTED** [review]

File: components/Ink/src/private/MDCLegacyInkLayer.m (hunk 2 above)
Comment (maicki) [reply components/Ink/src/private/MDCLegacyInkLayer.m]: Thanks for catching. Done

**Review by yarneo — APPROVED** [review]

**Pull request merged**